### PR TITLE
Added RMSProp and Adadelta optimizers

### PR DESCRIPTION
**Pull request by MarcCote**
This PR adds two new optimizers RMSProp and Adadelta.

### CODE REVIEW

File: smartlearner/optimizers/adadelta.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        Parameters
+        ----------
+        loss : `Loss`
+            The value (a theano computation graph) to optimize.
+        decay : float, optional
+            Decay rate (related to the window of the moving average).
+        epsilon : float, optional
+            Used to prevent a division by zero in early iterations.
+
+        Refrences
```
Comment (ppoulin91) [inline smartlearner/optimizers/adadelta.py]: Typo

File: smartlearner/optimizers/rmsprop.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        loss : `Loss`
+            The value (a theano computation graph) to optimize.
+        lr : float
+            Learning rate.
+        decay : float, optional
+            Decay rate (related to the window of the moving average).
+        epsilon : float, optional
+            Used to prevent a division by zero in early iterations.
+
+        Refrences
```
Comment (ppoulin91) [inline smartlearner/optimizers/rmsprop.py]: Typo

File: smartlearner/optimizers/adadelta.py (hunk 1 above)
Comment (MarcCote) [inline smartlearner/optimizers/adadelta.py]: thx

File: smartlearner/optimizers/rmsprop.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        return self._updates
+
+    def _get_directions(self):
+        """ Produces descending directions. """
+        directions = OrderedDict()
+
+        for i, (param, gradient) in enumerate(self.loss.gradients.items()):
+            # mean_squared_grad := \sum dc * g_{t-1}^2 + (1-dc) g_t^2
+            param_name = param.name if param.name is not None else str(i)
+            mean_squared_grad = sharedX(param.get_value() * 0., name='mean_squared_grad_' + param_name)
```
Comment (ppoulin91) [inline smartlearner/optimizers/rmsprop.py]: param.get_value() \* 0. ?

Comment (ppoulin91) [inline smartlearner/optimizers/rmsprop.py]: (I know it must be there for a reason, I just don't understand)

Comment (MarcCote) [inline smartlearner/optimizers/rmsprop.py]: Equivalent to `np.zeros_like(param.get_value())` but in a shorter way. I'll change it so it is clearer.

Comment (ppoulin91) [inline smartlearner/optimizers/rmsprop.py]: Ah! Thanks

**Comment by MarcCote** [comment]
@ppoulin91 if it is all good with you, go ahead and merge it Then start using them :D!

**Pull request merged**